src/evmTest/scala/io/iohk/ethereum/vm/utils/EvmTestEnv.scala

```diff
@@ -0,0 +1,142 @@
+trait EvmTestEnv {
+
+  import EvmTestEnv._
+
+  private var contractsAddresses: Map[String, Address] = Map.empty
+  private var contractsAbis: Map[String, Seq[ABI]] = Map.empty
+
+  private var internalWorld = MockWorldState()
+
+  val defaultContractCreator = createAccount(balance = 1000)
+  val defaultSender = createAccount(balance = 1000)
+
+  def world: MockWorldState = internalWorld
+
+  def contract(name: String): Contract = {
+    new Contract(name, contractsAddresses(name))
+  }
+
+  def createAccount(balance: BigInt = 0): Address = {
+    val newAddress = Address(Random.nextLong())
+    internalWorld = world.saveAccount(newAddress, Account.Empty.copy(balance = balance))
+    newAddress
+  }
+
+  def deployContract(name: String,
+                     creatorAddress: Address = defaultContractCreator,
+                     value: BigInt = 0,
+                     gasLimit: BigInt = BigInt(2).pow(256) - 1,
+                     gasPrice: BigInt = 1,
+                     constructorArgs: Seq[Any] = Nil): (ProgramResult[MockWorldState, MockStorage], Contract) = {
+    val creator = world.getAccount(creatorAddress).get
+
+    val contractAddress = {
+      val hash = crypto.kec256(rlp.encode(RLPList(creatorAddress.bytes, creator.nonce)))
+      Address(hash.takeRight(Address.Length))
+    }
+
+    val contractInitCode = Utils.loadContractCodeFromFile(new File(s"$ContractsDir/$name.bin"))
+    val contractAbi = Utils.loadContractAbiFromFile(new File(s"$ContractsDir/$name.abi"))
+
+    val payload = constructorArgs.map(parseArg).foldLeft(contractInitCode)(_ ++ _)
+
+    val tx = MockVmInput.transaction(creatorAddress, payload, value, gasLimit, gasPrice)
+    val bh = MockVmInput.blockHeader
+
+    val context = ProgramContext[MockWorldState, MockStorage](tx, bh, world)
+    val result = VM.run(context)
+
+    contractsAbis += (name -> contractAbi.right.get)
+    contractsAddresses += (name -> contractAddress)
+
+    internalWorld = result.world
+      .saveAccount(creatorAddress, creator.increaseNonce)
+      .saveAccount(contractAddress, Account(0, value, Account.EmptyStorageRootHash, kec256(result.returnData)))
+      .saveCode(contractAddress, result.returnData)
+
+    (result, new Contract(name, contractAddress))
+  }
+
+  private def parseArg(arg: Any): ByteString = arg match {
+    case b: ByteString => UInt256(b).bytes
+    case b: BigInt => UInt256(b).bytes
+    case a: Array[Byte] => UInt256(a).bytes
+    case i: Int => UInt256(i).bytes
+    case a: Address => UInt256(a.bytes).bytes
+    case other => throw new RuntimeException("Invalid call argument")
+  }
+
+  class Contract(val name: String, val address: Address) extends Dynamic {
+
+    def applyDynamic(methodName: String)(args: Any*): ContractMethodCall = {
+      callMethod(methodName)(args: _*)
+    }
+
+    def callMethod(methodName: String)(args: Any*): ContractMethodCall = {
+      if (methodName.contains("("))
+        callWithSignature(methodName)(args: _*)
```

Comment:
Yes, by default it finds ABI by parameter count, however if there are two overloaded methods with same parameter count one need to specify full signature to disambiguate. see: https://github.com/input-output-hk/etc-client/pull/97/files/8083de05eedc45c18a3e58b27328797d597b034b#diff-1c9763f46649c5ebc0ad7b9444ef8aa4R111